src/track/serato/markers.h

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <QByteArray>
+#include <QColor>
+#include <QDataStream>
+#include <QList>
+#include <memory>
+
+#include "util/types.h"
+
+// Forward declaration
+class Track;
+
+namespace mixxx {
+
+class SeratoMarkersEntry;
+typedef std::shared_ptr<SeratoMarkersEntry> SeratoMarkersEntryPointer;
+
+class SeratoMarkersEntry {
+  public:
+    enum class TypeId {
+        Unknown,
+        Cue,
+        Loop,
+    };
+
+    SeratoMarkersEntry(
+            bool hasStartPosition,
+            int startPosition,
+            bool hasEndPosition,
+            int endPosition,
+            QRgb color,
+            int type,
+            bool isLocked)
+            : m_color(color),
+              m_hasStartPosition(hasStartPosition),
+              m_hasEndPosition(hasEndPosition),
+              m_isLocked(isLocked),
+              m_startPosition(startPosition),
+              m_endPosition(endPosition),
+              m_type(type) {
+    }
+    ~SeratoMarkersEntry() = default;
+
+    QByteArray dump() const;
+    static SeratoMarkersEntryPointer parse(const QByteArray& data);
+
+    int type() const {
```

Comment:
typeCode?

Reply:
We could use an enum for this type code.

Reply:
We do, see the function below. It's just that I'd like to keep the original data in case they start adding new entry types that Mixxx doesn't know about.